pyproject.toml

```diff
@@ -34,3 +36,109 @@ build-backend = "poetry.core.masonry.api"
 
 [project]
 license = {text = "MIT License"}
+
+[tool.poetry.group.typing.dependencies]
+mypy = "1.9.0"
+
+[tool.bandit]
+exclude_dirs = ["tests"]
+
+[tool.coverage.run]
+source = ["upload_service_cl"]
+omit = [
+    "tests/conftest.py",
+]
+
+[tool.coverage.report]
+# Regexes for lines to exclude from consideration
+exclude_lines = [
+    # Don't complain if tests don't hit defensive assertion code:
+    "raise NotImplementedError",
+    "pragma: no cover",
+    # Coverage is not expected on pytest.fail() calls in tests
+    ".*pytest.fail.*",
+]
+fail_under = 80
+skip_covered = true
+show_missing = true
+
+[tool.mypy]
+python_version = "3.12"
+files = ["upload_service_cl"]
+enable_error_code = ["ignore-without-code"]
+check_untyped_defs = true
+allow_incomplete_defs = false
+warn_unused_configs = true
+no_implicit_optional = true
+warn_redundant_casts = true
+warn_unused_ignores = true
+warn_return_any = true
+
+[[tool.mypy.overrides]]
+ignore_missing_imports = true
+
+[tool.pytest.ini_options]
+# Make sure that warnings raise errors, to handle things like deprecations as soon as possible.
+filterwarnings = [
+    "error",  # Make sure that warnings raise errors, to handle things like deprecations as soon as possible.
+    # overrides for the errors: schemathesis and hypothesis
+    "ignore::DeprecationWarning:hypothesis_jsonschema",
+    "ignore::DeprecationWarning:schemathesis",
+    "ignore:.*custom tp_new.*in Python 3.14.*:DeprecationWarning", # This is an issue not resolved in the gcs lib yet
```

Comment:
I don't think we need those exceptions here

Reply:
You right